worker_node/job/send.js

```diff
@@ -141,8 +153,10 @@ class Send extends Base {
     // const baseTimestamp = this.config.timestamp;
     const baseTx = this.makeBaseTransaction();
     const timestampSet = new Set();
+    const targetTestEndTime = Date.now() + (this.config.duration * 1000); // MS
 
     for (let i = 0; i < this.config.numberOfTransactions; i++) {
+      const delayTime = (targetTestEndTime - Date.now()) / (this.config.numberOfTransactions - i);
       await delay(delayTime);
```

Comment:
Nice! Such an elegant solution! 👍 Could you just add a if (delayTime) > 0 condition check before calling delay()?

Reply:
Thanks! Done!